packages/appium/lib/extension/extension-config.js

```diff
@@ -113,11 +124,94 @@ export class ExtensionConfig {
+    if (this._listDataCache) {
+      return this._listDataCache;
+    }
+    const CommandClass = /** @type {ExtCommand<ExtType>} */ (commandClasses[this.extensionType]);
+    const cmd = new CommandClass({config: this, json: true});
+    const listData = await cmd.list({showInstalled: true, showUpdates: true});
+    this._listDataCache = listData;
+    return listData;
+  }
+
+  /**
+   * Displays non-"fatal" warnings for the extension
+   *
+   * This method uses Appium's logger, since the default extension logger uses the `error` log level.
+   *
    * @param {ExtManifest<ExtType>} extData
    * @param {ExtName<ExtType>} extName
+   * @returns {Promise<void>}
+   */
+  async displayConfigWarnings(extData, extName) {
+    const {appiumVersion, installSpec, installType, pkgName} = extData;
+
+    if (!_.isString(installSpec)) {
+      log.warn(
+        `${_.capitalize(
+          this.extensionType
+        )} "${extName}" (package \`${pkgName}\`) has an invalid or missing \`installSpec\` property in \`extensions.yaml\`; this may cause upgrades done via the \`appium\` CLI to fail.`
+      );
+    }
+
+    if (!INSTALL_TYPES.has(installType)) {
+      log.warn(
+        `${_.capitalize(
+          this.extensionType
+        )} "${extName}" (package \`${pkgName}\`) has an invalid or missing \`installType\` property in \`extensions.yaml\`; this may cause upgrades done via the \`appium\` CLI to fail.`
+      );
+      extData.installType = INSTALL_TYPE_NPM;
+    }
+    if (_.isString(appiumVersion) && !satisfies(APPIUM_VER, appiumVersion)) {
+      const listData = await this.getListData();
+
+      if (listData[extName]) {
+        const extListData =
+          /** @type {import('../cli/extension-command').InstalledExtensionListData} */ (
+            listData[extName]
+          );
+        const {unsafeUpdateVersion, updateVersion, upToDate} = extListData;
+        if (!upToDate) {
+          const upgradeText =
+            unsafeUpdateVersion === updateVersion
+              ? `v${updateVersion}`
+              : `v${updateVersion} or (potentially unsafe) v${unsafeUpdateVersion}`;
+          log.warn(
+            `${_.capitalize(
+              this.extensionType
+            )} "${extName}" (package \`${pkgName}\`) may be incompatible with the current version of Appium (v${APPIUM_VER}) due to its peer dependency on older Appium v${appiumVersion}. Please upgrade \`${pkgName}\` to ${upgradeText}.`
+          );
+        } else {
+          log.warn(
+            `${_.capitalize(
+              this.extensionType
+            )} "${extName}" (package \`${pkgName}\`) may be incompatible with the current version of Appium (v${APPIUM_VER}) due to its peer dependency on older Appium v${appiumVersion}. Please ask the developer of \`${pkgName}\` to update the peer dependency on Appium to ${APPIUM_VER}.`
+          );
+        }
+      }
+    } else if (!_.isString(appiumVersion)) {
+      log.warn(
+        `${_.capitalize(
```

Comment:
can't say i love prettier's choice of line breaks here

Reply:
yeah it is pretty aggressive about linebreaks in template literals when there's a function call.  it kind of makes sense, but I try not to care too much